xla/service/gpu/runtime/command_buffer_cmd.cc

```diff
@@ -1448,12 +1448,12 @@ absl::Status AllReduceCmd::Record(const Thunk::ExecuteParams& execute_params,
 
   // Today when recording collective operations into command buffers we always
   // use a sync mode and a stream id `0`.
-  TF_ASSIGN_OR_RETURN(NcclApi::NcclCommHandle comm,
+  TF_ASSIGN_OR_RETURN(NcclCommHandleWrapper comm_handle,
                       GetNcclComm(*execute_params.collective_params,
                                   *execute_params.collective_cliques,
                                   config().replica_groups, config().group_mode,
                                   /*stream_id=*/0, GetAsyncStreamKind()));
-
+  NcclApi::NcclCommHandle comm = comm_handle.comm_handle;
```

Comment:
I wonder can you do `TF_ASSIGN_OR_RETURN(auto& [comm, _], ...`)? If not that's certainly something to fix in macro definition (some time later)

Reply:
it doesnt work for me, i have seen this for a pair or tuple, didnt know it would work for well defined struct.